curvefs/src/mds/topology/topology_manager.cpp

```diff
@@ -735,25 +737,91 @@ bool TopologyManager::CreateCopysetNodeOnMetaServer(
 }
 
 void TopologyManager::ClearCopysetCreating(PoolIdType poolId,
-    const std::set<CopySetIdType> &copysets) {
-    for (const auto &id : copysets) {
-        topology_->RemoveCopySetCreating(CopySetKey(poolId, id));
+                                           CopySetIdType copysetId) {
+    topology_->RemoveCopySetCreating(CopySetKey(poolId, copysetId));
+}
+
+TopoStatusCode TopologyManager::CreateEnoughCopyset() {
+    int avaibleCopysetNum = topology_->GetAvailableCopysetNum();
+    int copysetNumTotal = topology_->GetCopySetsInCluster().size();
+
+    int createNum = option_.createCopysetNumber - avaibleCopysetNum;
+    if (createNum <= 0) {
+        return TopoStatusCode::TOPO_OK;
+    }
+
+    TopoStatusCode ret = TopoStatusCode::TOPO_OK;
```

Comment:
You can add some notes here to explain the logic.

Reply:
done